src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/RescindRecordControllerTest.kt

```diff
@@ -0,0 +1,271 @@
+    val fileName = "file.msg"
+    val decisionDetails = "blah blah again"
+    val rescindRecordId = ::RescindRecordId.random()
+    val emailId = ::DocumentId.random()
+    val rescindRecord = RescindRecord(rescindRecordId, recallId, 1, userId, OffsetDateTime.now(), emailId, "details", LocalDate.now())
+    val savedRecordSlot = slot<RescindRecord>()
+
+    every { recall.rescindRecords } returns setOf(rescindRecord)
+    every { recallRepository.getByRecallId(recallId) } returns recall
+    every { tokenExtractor.getTokenFromHeader(bearerToken) } returns TokenExtractor.Token(userId.toString())
+    every {
+      documentService.scanAndStoreDocument(
+        recallId,
+        userId,
+        documentBytes,
+        DocumentCategory.RESCIND_DECISION_EMAIL,
+        fileName
+      )
+    } returns Success(emailId)
+    every { rescindRecordRepository.save(capture(savedRecordSlot)) } returns rescindRecord
+
+    val decisionRequest = RescindDecisionRequest(
+      false,
+      decisionDetails, LocalDate.now(),
+      documentBytes.encodeToBase64String(), fileName
+    )
+
+    val response =
+      underTest.decideRescindRecord(recallId, rescindRecordId, decisionRequest, bearerToken)
+
+    assertThat(savedRecordSlot.captured.approved, equalTo(false))
+    assertThat(savedRecordSlot.captured.createdDateTime, !equalTo(savedRecordSlot.captured.lastUpdatedDateTime))
+    assertThat(savedRecordSlot.captured.decisionDetails, equalTo(decisionDetails))
+    assertThat(savedRecordSlot.captured.decisionEmailId, present())
+    assertThat(response.statusCode, equalTo(HttpStatus.OK))
+    assertThat(
+      response.body,
+      equalTo(
+        rescindRecordId
+      )
+    )
+  }
+
+  @Test
+  fun `403 error thrown when requesting a new rescind if one is already in progress`() {
+    val recall = mockk<Recall>()
+    val documentBytes = "a document".toByteArray()
+    val fileName = "email.msg"
+    val mockRecord = mockk<RescindRecord>()
+    val rescindRecordId = ::RescindRecordId.random()
+    val details = "blah blah"
+
+    every { recall.rescindRecords } returns setOf(mockRecord)
+    every { recallRepository.getByRecallId(recallId) } returns recall
+    every { tokenExtractor.getTokenFromHeader(bearerToken) } returns TokenExtractor.Token(userId.toString())
+    every { mockRecord.id() } returns rescindRecordId
+    every { mockRecord.hasBeenDecided() } returns false
+    every { mockRecord.version } returns 1
+
+    val request = RescindRequestRequest(
+      details, LocalDate.now(),
+      documentBytes.encodeToBase64String(), fileName
+    )
+
+    val response2 = underTest.createRescindRecord(recallId, request, bearerToken)
+
+    assertThat(response2.statusCode, equalTo(HttpStatus.FORBIDDEN))
+
+    verify { documentService wasNot Called }
+    verify { rescindRecordRepository wasNot Called }
+  }
+
+  @Test
+  fun `NotFoundException thrown if rescind record does not exist when deciding`() {
```

Comment:
Minor but I'm wondering what we get in response to recallId not found on e.g. the POST request.

Reply:
Added tests